Automattic/newspack-plugin
# feat(plugins): add url redirect for wc subscription renewals

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://app.asana.com/0/1206274567818686/1208437232706180/f

This PR adds handling for ~~a `np_renewal` param~~ a new `my-account/my-renewals` endpoint to redirect readers to the first available pending renewal checkout, otherwise the subscriptions my account page:

![Screenshot 2024-11-05 at 16 39 08](https://github.com/user-attachments/assets/733544ac-979d-40f9-9276-fe392227dfa6)

### How to test the changes in this Pull Request:

1. As a reader, purchase multiple subscriptions
2. As admin, go to any subscription edit page for one of these subscriptions via WooCommerce > Subscriptions
3. Create a renewal order for the subscription
![Screenshot 2024-11-05 at 16 41 21](https://github.com/user-attachments/assets/306ebe26-daa6-4d02-a938-d59d082604b7)
4. As the logged in reader, visit the `my-account/my-renewals` endpoint of your test site: `site.url/my-account/my-renewals`
5. Confirm you are redirected to checkout for the renewal order in question
6. As admin again, create another renewal order for another subscription belonging to this reader (there should now be two subscriptions that need payment)
7. As reader again, visit the url from step 3 again
8. This time confirm you are directed to the my account subscriptions page
![Screenshot 2024-11-06 at 18 19 15](https://github.com/user-attachments/assets/6355e8cd-bc7e-4c21-90c1-7514f1a55671)
9. Complete checkout to re-activate the subscription. Then do the same for the second pending renewal (All subscriptions should be active).
10. Visit the url from step 3 again.
11. Confirm you are directed to the my account subscriptions page again
12. Log out
13. Visit the url from step 3 again.
14. Confirm you are presented with a login form
15. Login and confirm you are redirected to the my account subscriptions page once again.

### Other information:
